tests/pipelines/test_pipelines_common.py

```diff
@@ -616,7 +615,7 @@ def test_save_load_optional_components(self, expected_max_difference=1e-4):
         inputs = self.get_dummy_inputs(torch_device)
         output_loaded = pipe_loaded(**inputs)[0]
 
-        max_diff = np.abs(to_np(output) - to_np(output_loaded)).max()
+        max_diff = numpy_cosine_similarity_distance(to_np(output).flatten(), to_np(output_loaded).flatten())
```

Comment:
https://github.com/huggingface/diffusers/pull/4812/files#r1307509747 would be better